047-TrafficControlWithDapr/Student/Challenge-00.md

```diff
@@ -0,0 +1,356 @@
+
+*It's helpful to refer back to the preceding sequence diagram as you progress through the challenges.*
+
+### Prevent port collisions
+
+For most of the challenges, you'll run the microservices in the solution on your local machine. To prevent port collisions, all services will listen on a different HTTP port. When running with Dapr, you need additional ports for HTTP and gRPC communication between the sidecar services. By default, these ports are `3500` and `50001`. However, you'll use different port numbers for each service to prevent collisions. Please closely follow the instructions so that your microservices use the following ports for their Dapr sidecars:
+
+| Service                    | Application Port | Dapr sidecar HTTP port | Dapr sidecar gRPC port |
+| -------------------------- | ---------------- | ---------------------- | ---------------------- |
+| TrafficControlService      | 6000             | 3600                   | 60000                  |
+| FineCollectionService      | 6001             | 3601                   | 60001                  |
+| VehicleRegistrationService | 6002             | 3602                   | 60002                  |
+
+Use the ports specified in the preceding table *whether* using the DIY or step-by-step approach.
+
+You'll specify the ports from the command-line when starting a service with the Dapr CLI using the following command-line arguments:
+
+- `--app-port`
+- `--dapr-http-port`
+- `--dapr-grpc-port`
+
+### Step 1. Install pre-requisites
+
+1.  To start, you'll need access to an Azure Subscription:
+
+    - If you don't have one, [Sign Up for an Azure account](https://azure.microsoft.com/en-us/free/).
+    - If you already have an Azure account, make sure you have at least [Contributor access instructions](https://docs.microsoft.com/azure/role-based-access-control/check-access)) for the resource group in which you'll provision Azure resources.
+        
+    *Your IT organization may provide you access to an Azure resource group, but not the entire subscription. If that's the case, take note of that resource group name and make sure you have `Contributor` access to it, using the instructions mentioned above.*
+  
+1.  Install all the pre-requisites listed below and make sure they're working correctly:
+
+    - Git ([download](https://git-scm.com/))
+    - .NET 5 SDK ([download](https://dotnet.microsoft.com/download/dotnet/5.0))
+    - Visual Studio Code ([download](https://code.visualstudio.com/download)) with the following extensions installed:
+      - [C#](https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.csharp)
+      - [REST Client](https://marketplace.visualstudio.com/items?itemName=humao.rest-client)
+    - Docker for desktop ([download](https://www.docker.com/products/docker-desktop))
+    - Dapr CLI and Dapr runtime ([instructions](https://docs.dapr.io/getting-started/install-dapr-selfhost/))
+    - Install Azure CLI ([instructions]())
```

Comment:
Busted link to Azure CLI

Reply:
Same as on home page... points to Bicep installer for Linux instead of Linux Azure CLI install page.